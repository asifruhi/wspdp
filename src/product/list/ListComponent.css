.products-list {
	list-style: none;
	padding-left: 0;
}

.products-list .product {
	margin-bottom: 3px;
	text-align: center;
	position: relative;
}

.products-list .product .product-name {
	line-height: 1.5em;
  padding: 10px;

	font-weight: 500;
	text-align: left;
  background: white;
  opacity: 0.6;

  position: absolute;
  top: 0;
  left: 0;
  right: 0;
}

.products-list .product .product-price {
  position: absolute;
  left: 15px;

  font-weight: 500;
  line-height: 1.5em;
  background: black;
  opacity: 0.7;

  padding-left: 5px;
  padding-right: 5px;
}

.products-list .product .regular {
	color: white;
	bottom: 4em;
}

.products-list .product .selling {
	color: red;
	bottom: 2.2em;
}

@media screen and (max-width: 35.5em) {
	.products-list .product img {
		width: 100%;
	}
}

/*small*/
@media screen and (min-width: 35.5em) and (max-width: 48em) {
	.products-list .product img {
		width: 100%;
	}
}

@media screen and (min-width: 48em) {
	.products-list {
  	padding-left: 10px;
  }

	.products-list .product img {
		width: 98%;
	}
}
