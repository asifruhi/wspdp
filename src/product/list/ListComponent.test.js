import React from 'react';
import renderer from 'react-test-renderer';

import ListComponent from './ListComponent';

it('renders products list correctly', () => {
	const products = [{
		id: "metallic-square-lanterns-d5161",
		name: "Metallic Square Lanterns",
		priceRange:{
			regular: {high:149,low:24},
			selling:{high:111,low:18},
			type:"special"
		},
		thumbnail:{
			href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201810/0031/metallic-square-lanterns-m.jpg"
		}
	}, {
    id: "terrace-lanterns-d5183",
    name: "Terrace Lanterns",
    priceRange:{
      regular: {high:44,low:39},
      selling:{high:44,low:29},
      type:"special"
    },
    thumbnail:{
      href: "https://www.westelm.com/weimgs/ab/images/wcm/products/201808/0047/gold-glass-candleholders-m.jpg"
    }
  }]
  const tree = renderer
    .create(<ListComponent title="Some Products" products={products}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});