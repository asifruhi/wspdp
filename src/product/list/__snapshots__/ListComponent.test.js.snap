// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders products list correctly 1`] = `
<div>
  <h3
    className="page-title"
    tabIndex="2"
  >
    Some Products
  </h3>
  <ul
    className="pure-g products-list"
    role="grid"
  >
    <li
      className="pure-u-1 pure-u-md-1-2 pure-u-lg-1-3 pure-u-xl-1-4 product"
      role="gridcell"
    >
      <a
        href="/products/metallic-square-lanterns-d5161"
        role="link"
      >
        <div>
          <img
            alt="image for Metallic Square Lanterns"
            src="https://www.westelm.com/weimgs/ab/images/wcm/products/201810/0031/metallic-square-lanterns-m.jpg"
          />
          <div
            aria-label="Product Name is Metallic Square Lanterns"
            className="product-name"
          >
            Metallic Square Lanterns
          </div>
          <div>
            <div
              aria-label="Regular price range is $24 to $149"
              className="regular product-price"
            >
              $24
               - 
              $149
            </div>
            <div
              aria-label="Selling price range is $18 to $111"
              className="selling product-price"
            >
              $18
               - 
              $111
            </div>
          </div>
        </div>
      </a>
    </li>
    <li
      className="pure-u-1 pure-u-md-1-2 pure-u-lg-1-3 pure-u-xl-1-4 product"
      role="gridcell"
    >
      <a
        href="/products/terrace-lanterns-d5183"
        role="link"
      >
        <div>
          <img
            alt="image for Terrace Lanterns"
            src="https://www.westelm.com/weimgs/ab/images/wcm/products/201808/0047/gold-glass-candleholders-m.jpg"
          />
          <div
            aria-label="Product Name is Terrace Lanterns"
            className="product-name"
          >
            Terrace Lanterns
          </div>
          <div>
            <div
              aria-label="Regular price range is $39 to $44"
              className="regular product-price"
            >
              $39
               - 
              $44
            </div>
            <div
              aria-label="Selling price range is $29 to $44"
              className="selling product-price"
            >
              $29
               - 
              $44
            </div>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
